daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/Ast.scala

```diff
@@ -488,67 +488,101 @@ object Ast {
   // Definitions
   //
 
-  sealed abstract class Definition extends Product with Serializable
+  sealed abstract class AbstractDefinition[+E] extends Product with Serializable
```

Comment:
I agree the names is off. We should find another one.
However I want to reserve `Definition`  for the type alias `AbstractDefinition[Expr]`.
Any other suggestion ?

Reply:
Not really unfortunately. Naming is hard :crying_cat_face:

Reply:
> I would have called this `GenDefinition`, if that helps.

:+1:

Reply:
> I would also have called the type parameter `Expr` instead of `E`, to keep the diff smaller. But that's a matter of priorities and taste I guess.

If we do this  we get "nice" warnings like this:
```
daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/Ast.scala:679: warning: type parameter Expr defined in class GenModule shadows class Expr defined in object Ast. You may want to rename your type parameter, or possibly remove it.
  case class GenModule[Expr] private[Ast] (
```
What should I do ?
1 - find another renaming (e.g. "Expression")
2 - add a least 11 `@silent` anotations

Reply:
> What should I do ?

Let's stick with `E` then. Minimizing diff noise should not be done at the expense of creating compiler warnings, even in my world. 😄